packages/hs-react-ui/src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,279 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import styled, { StyledComponentBase } from 'styled-components';
+import Icon from '@mdi/react';
+import { mdiCheck, mdiClose, mdiMenuDown, mdiMenuUp } from '@mdi/js';
+import { readableColor } from 'polished';
+
+import Button from '../Button';
+import { ButtonContainer } from '../Button/Button';
+import colors from '../../enums/colors';
+import timings from '../../enums/timings';
+
+const Container = styled.div<{ elevation: number }>`
+  ${({ elevation }) => {
+  const elevationYOffset = elevation && elevation >= 1 ? (elevation - 1) * 0.5 + 0.1 : 0;
+  const elevationBlur = elevation && elevation >= 1 ? (elevation / 16) * 0.1 + 0.3 : 0;
+  return `
+    width: fit-content;
+    transition: filter ${timings.slow};
+    filter: drop-shadow(0rem ${elevationYOffset}rem ${elevationBlur}rem rgba(0,0,0,${0.6 - elevation * 0.1}));
+  `;
+}}
+  
+`;
+const ValueContainer = styled(ButtonContainer)`
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-end;
+  align-items: center;
+  
+  line-height: initial;
+  width: 15rem;
+  padding: 0.5rem;
+`;
+const ValueIconContainer = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: flex-end;
+  
+  width: 3rem;
+`;
+
+const ValueItem = styled.div`
+  width: 100%;
+  text-align: left;
+`;
+
+const OptionsContainer = styled.div`
+  background: white;
+  position: absolute;
+  max-height: 10rem;
+  overflow-y: scroll;
+  width: 15rem;
+  border: 0.5px solid ${colors.blueCharcoal25};
+`;
+
+const OptionItem = styled.div`
+  height: 2rem;
+  display: flex;
+  align-items: center;
+  
+  font-family: Montserrat,Roboto,sans-serif;
+  
+  &:focus,
+  &:hover {
+    background: ${colors.blueCharcoal50};
+    cursor: pointer;
+    outline: none;
+  }
+`;
+const CheckContainer = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  
+  padding-right: 0.2rem;
+  width: 2rem;
+`;
+
+export interface DropdownProps {
+  StyledContainer?: string & StyledComponentBase<any, {}>;
+  StyledValueContainer?: string & StyledComponentBase<any, {}>;
+  StyledValueItem?: string & StyledComponentBase<any, {}>;
+  StyledOptionsContainer?: string & StyledComponentBase<any, {}>;
+  StyledOptionItem?: string & StyledComponentBase<any, {}>;
+
+  clearable?: boolean,
+  elevation?: number,
+  multi?: boolean,
+  name: string,
+  onClear?: () => void,
+  onSelect?: (selected: string | Array<string>) => void,
+  options: Array<string>,
+  tabIndex?: number,
+  values?: Array<string>,
```

Comment:
for my own education, what's the difference between this and `string[]`?

Reply:
https://stackoverflow.com/questions/38239579/what-is-the-difference-arraystring-and-string

TL;DR Nothing, except Array<string> can be extended, string[] can't - that's why I just tend to prefer using Array<string>

Reply:
neat